packages/vitest/src/integrations/wait.ts

```diff
@@ -52,6 +53,10 @@ export function waitFor<T>(callback: WaitForCallback<T>, options: number | WaitF
     }
 
     const checkCallback = () => {
+      // use fake timers
+      if (globalThis.setTimeout !== setTimeout)
```

Comment:
I haven't found a better way to tell if `useFakeTimer` is being used or not

Reply:
I think fake timer has `_clock` property or something like that

Reply:
Sure, we can add it to vi object